apps/parsley/src/utils/searchLogs/index.ts

```diff
@@ -13,34 +13,53 @@ interface searchOptions {
   processedLogLines: ProcessedLogLines;
 }
 
+/**
+ * Searches through the processed log lines for matches to the provided regular expression.
+ * @param options Object containing search parameters and helpers.
+ * @param options.searchRegex The regular expression to search for in the logs.
+ * @param options.upperBound Inclusive upper bound for the search.
+ * @param options.lowerBound Inclusive lower bound for the search.
+ * @param options.getLine Function to get the raw log string at the given index.
+ * @param options.processedLogLines Processed log lines that are currently rendered.
+ * @returns An array of sorted raw log indices that match the search criteria. SkippedLines are not included in the result.
+ */
 const searchLogs = (options: searchOptions): number[] => {
   const { getLine, lowerBound, processedLogLines, searchRegex, upperBound } =
```

Comment:
Maybe we should set the upperBound to make the if statements less tedious. even like `Number.MAX_VALUE`